### PR TITLE
new endpoint to get a list of service accounts

### LINKED ISSUE #3045: Create new List endpoint for SA
**User Story**
Create new endpoint to get a list of service accounts for the project.

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: new endpoint to get a list of service accounts

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3045

```release-note
a new endpoint for getting ServiceAccounts for projects: 'GET /api/v1/projects/{project_id}/serviceaccounts'
```

### CODE REVIEW

**Comment by zreigz** [comment]
@p0lyn0mial PTAL

File: api/pkg/handler/v1/serviceaccount/sa.go [hunk 1]
```diff
@@ -59,6 +59,54 @@ func CreateEndpoint(projectProvider provider.ProjectProvider, serviceAccountProv
...
+
+		saList, err := serviceAccountProvider.List(userInfo, project, nil)
+		if err != nil {
+			return nil, common.KubernetesErrorToHTTPError(err)
+		}
+
+		var errorList []string
+		response := make([]*apiv1.ServiceAccount, 0)
+		for _, sa := range saList {
+			internalSA := convertInternalServiceAccountToExternal(sa)
```
Comment (p0lyn0mial) [inline api/pkg/handler/v1/serviceaccount/sa.go]: nit: change `internalSA` to `externalSA`

**Review by p0lyn0mial — COMMENTED** [review]
/lgtm
/approve

File: api/pkg/handler/v1/serviceaccount/sa.go (hunk 1 above)
Comment (p0lyn0mial) [reply api/pkg/handler/v1/serviceaccount/sa.go]: feel free to change it in the next pull.

Comment (zreigz) [reply api/pkg/handler/v1/serviceaccount/sa.go]: I will, thanks

**Pull request merged**